README.md

```diff
@@ -104,21 +104,26 @@ Run tests with [Tox][]
 poetry run tox
 ```
 
-To run integration tests, you need a git repository set up with token access and
-a branch named after your git repo username (stripped of any @domain if your
-username is an email address).
+To run integration tests, you need:
 
-Make sure the artifacts/repo section of ~/.softpack/core/config.yml is
+- a git repository
+- an access token for the git repository
+  - for GitLab, this requires the "developer" role and the "write_repository" scope
+  - for GitHub, this requires read-write access to repository contents
+- a branch to run the tests on (not called "main")
```

Comment:
It needs to be named after your git repo username because IIRC there's code that uses your username to decide what branch to test with (to avoid accidentally running against main and avoid the need to add something to config)

Reply:
yes, there's code to prevent accidentally running the integration tests on `main`. I still don't understand what is meant by "git repo username", though, in contrast to e.g. "your sanger username"

my point is: does this need to match some property of the git repo beyond "being a branch that exists", or is it just "some unique string used to avoid your integration-test commits colliding with other people's branches or with real work"?

Reply:
One of the existing config options is your GitHub username, needed for it to commit to your GitHub repo. So to pick a branch name that won't conflict with potential other people using the same test repo, it uses your GitHub username as the branch name.

Reply:
(or gitlab in our case IIRC)

Reply:
ahhh! I didn't realise it uses `username` for authentication – it seemed to work fine with a personal access token in `writer` and any old string in `username`, on both github and gitlab. I guess if you're typing your password in directly, it would need to match your username on that platform. that makes sense!